### ISSUE TITLE
tests are silently failing in CI

### ISSUE THREAD

**Issue by alexwlchan** [wellcomecollection/wellcomecollection.org#7662]
Quoting [Paul in Slack](https://wellcome.slack.com/archives/C3TQSF63C/p1645029469597069):

> The test stages in Experience fail silently on buildkite.  e.g. see the test content stage here https://buildkite.com/wellcomecollection/experience/builds/5721#fd698ae6-77eb-42e6-bb31-e46799205505

**Comment by paul-butcher** [comment]
My first suspicion is that it is to do with this (and a similar thing in catalogue):
`process.exit called with "0"` when a server is closed during the last test.

So the tests could all fail, but it will still return a zero exit code as long as that test runs.  It may also mean that some tests are not being run (if `process.exit(0)` happens when there are still outstanding tests)

**Comment by paul-butcher** [comment]
Yes. I have commented out most of that file in catalogue and forced a failure.

**Comment by paul-butcher** [comment]
It would appear that the mocking of process.exit is not working.

If I call process.exit in the test file, it uses the mock, but when it calls server.close, which calls process.exit it calls the real process.exit.

**Comment by paul-butcher** [comment]
That turns out to be when this mocking was introduced.
